charts/linkerd2-multicluster-remote-setup/templates/gateway.yaml

```diff
@@ -0,0 +1,95 @@
+---
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: {{ .Values.gatewayNamespace }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nginx-configuration
+  namespace: {{.Values.gatewayNamespace}}
+data:
+  nginx.conf: |-
+    events {
+    }
+    stream {
+      server {
+          listen     {{.Values.incomingPort}};
+          proxy_pass 127.0.0.1:{{.Values.proxyOutboundPort}};
+      }
+    }
+    http {
+      server {
+          listen     {{.Values.probePort}};
+          location {{.Values.probePath}} {
+            access_log off;
+            return 200 "healthy\n";
+          }
+      }
+    }
+---    
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{.Values.gatewayName}}
+  name: {{.Values.gatewayName}}
+  namespace: {{.Values.gatewayNamespace}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.Values.gatewayName}}
+  template:
+    metadata:
+      annotations:
+        linkerd.io/inject: enabled
```

Comment:
Do we want to use the proxy partial for this instead of relying on the injector?

Reply:
So if we want to use the proxy partials I think it will be best to bring that into the linkerd2 chart. Or else, we need to add all the config that is exposed to the proxy, like ports, etc etc. 

My assumption here is that we install that into a cluster that already has Linkerd installed and running in the specified namespace.  @alpeb you have better understanding of Helm charts. What do you think ?

Reply:
You could make this an addon, disabled by default, and put everything under `charts/add-ons/multicluster-remote-setup`. Just like `charts/add-ons/tracing` you can require partials. The global configs in linkerd's main `values.yaml` (which include the proxy configs) will be available in your addon, without having to redeclare them in your addon's `values.yaml`.

Reply:
I think a question here is if we want the gateway to be "pre-injected" with the proxy partial like the other control plane components, or if we just want to rely on the proxy injector.  I don't have a good sense for which of these options is best, but naively, using the proxy-injector seems simpler.

Reply:
I think we've got injection disabled in `linkerd`, so there's some implications on where we want this to actually live and run.